src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,266 @@
+
+
+def _check_health_environment(ignore_errors, yes):
+    from azure.cli.core.util import in_cloud_console
+    if in_cloud_console():
+        print_warning("Environment checks are not supported in Azure Cloud Shell.")
+        return
+
+    _get_docker_status_and_version(ignore_errors, yes)
+    _get_cli_version()
+    _get_helm_version(ignore_errors)
+
+
+# Checks for the connectivity
+
+
+# Check DNS lookup
+def _get_registry_status(login_server, ignore_errors):
+    import socket
+
+    registry_ip = ""
+
+    try:
+        registry_ip = socket.gethostbyname(login_server)
+    except socket.gaierror:
+        pass
+
+    if not registry_ip:
+        from ._errors import CONNECTIVITY_DNS_ERROR
+        _handle_error(CONNECTIVITY_DNS_ERROR.format_error_message(login_server), ignore_errors)
+        return False
+
+    print_pass("DNS lookup to {} at IP {}".format(login_server, registry_ip))
+
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+    challenge = requests.get('https://' + login_server + '/v2/', verify=(not should_disable_connection_verify()))
+
+    if challenge.status_code == 403:
+        from ._errors import CONNECTIVITY_FORBIDDEN_ERROR
+        _handle_error(CONNECTIVITY_FORBIDDEN_ERROR.format_error_message(login_server), ignore_errors)
+        return False
+    return True
+
+
+def _get_endpoint_and_token_status(cmd, login_server, ignore_errors):
+    from ._errors import CONNECTIVITY_CHALLENGE_ERROR, CONNECTIVITY_AAD_LOGIN_ERROR, \
+        CONNECTIVITY_REFRESH_TOKEN_ERROR, CONNECTIVITY_ACCESS_TOKEN_ERROR
+
+    # Check access to login endpoint
+    url = 'https://' + login_server + '/v2/'
+    result_from_token = _get_aad_token(cmd.cli_ctx, login_server, False, is_diagnostics_context=True)
+
+    if isinstance(result_from_token, ErrorClass):
+        if result_from_token.error_title == CONNECTIVITY_CHALLENGE_ERROR.error_title:
+            _handle_error(result_from_token, ignore_errors)
+            return
+
+        print_pass("Challenge endpoint {}".format(url))
+
+        if result_from_token.error_title == CONNECTIVITY_AAD_LOGIN_ERROR.error_title:
+            _handle_error(result_from_token, ignore_errors)
+            return
+
+        if result_from_token.error_title == CONNECTIVITY_REFRESH_TOKEN_ERROR.error_title:
+            _handle_error(result_from_token, ignore_errors)
+            return
+
+        print_pass("Obtain refresh token for registry '{}'".format(login_server))
+
+        if result_from_token.error_title == CONNECTIVITY_ACCESS_TOKEN_ERROR.error_title:
+            _handle_error(result_from_token, ignore_errors)
+            return
+
+        print_pass("Obtain access token for registry '{}'".format(login_server))
+
+        return
+
+    # If return is not of type ErrorClass, then it is the token
+    print_pass("Challenge endpoint {}".format(url))
+    print_pass("Obtain refresh token for registry '{}'".format(login_server))
```

Comment:
nit: consider `Fetch` or `Get` instead of `Obtain`

Reply:
Fetch sounds good :)